staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -940,6 +953,9 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 							allErrs = append(allErrs, field.Invalid(fldPath.Child("x-kubernetes-validations").Index(i).Child("rule"), schema.XValidations[i], cr.Error.Detail))
 						}
 					}
+					if cr.TransitionRule && ssv.unmergeable() {
```

Comment:
we don't want to merge this PR and allow `oldSelf` rules to be persisted until support for `oldSelf` is actually implemented, so make this forbid for now:
```go
if cr.TransitionRule {
  if ssv.unmergeable() {
    // ... append "oldSelf cannot be used in an unmergeable portion of the schema" error
  } else {
    // ... append "not yet implemented" error
  }
}
```

The unit tests using oldSelf in unmergeable schemas should expect the "cannot be used in an unmergeable portion" error, and the ones that would otherwise work can expect a "not yet implemented" error. When oldSelf is implemented, we'll drop that error and that expectation from those unit tests.